### PR TITLE
update validators

### PR DESCRIPTION
**Pull request by BelehovEgor**

### CODE REVIEW

File: src/CompanyService.Business/Commands/Department/GetDepartmentCommand.cs [hunk 1]
```diff
@@ -16,173 +20,185 @@
...
-                _logger.LogError(exc, loggerMessage);
-            }
-
-            errors.Add(message);
-
-            return new();
-        }
+      if (userIds == null || !userIds.Any())
+      {
+        return new();
```
Comment (djino97) [inline src/CompanyService.Business/Commands/Department/GetDepartmentCommand.cs:43]: may be write return null?

**Review by djino97 — APPROVED** [review]

File: src/CompanyService.Business/Commands/Department/GetDepartmentCommand.cs (hunk 1 above)
Comment (BelehovEgor) [reply src/CompanyService.Business/Commands/Department/GetDepartmentCommand.cs]: fixed

**Review by MariyaSin — APPROVED** [review]

File: src/CompanyService.Data/PositionRepository.cs [hunk 2]
```diff
@@ -1,112 +1,100 @@
...
-    /// <inheritdoc cref="IPositionRepository"/>
-    public class PositionRepository : IPositionRepository
+  /// <inheritdoc cref="IPositionRepository"/>
+  public class PositionRepository : IPositionRepository
+  {
+    private readonly IDataProvider _provider;
+    private readonly IHttpContextAccessor _httpContextAccessor;
+
+    public PositionRepository(
+        IDataProvider provider,
```
Comment (MariyaSin) [inline src/CompanyService.Data/PositionRepository.cs:22]: tab

File: src/CompanyService.Data/PositionRepository.cs [hunk 3]
```diff
@@ -1,112 +1,100 @@
...
+        IHttpContextAccessor httpContextAccessor)
+    {
+      _provider = provider;
+      _httpContextAccessor = httpContextAccessor;
+    }
+
+    public DbPosition Get(Guid positionId)
+    {
+        return _provider.Positions.FirstOrDefault(d => d.Id == positionId);
```
Comment (MariyaSin) [inline src/CompanyService.Data/PositionRepository.cs:31]: tab

File: src/CompanyService.Data/PositionRepository.cs [hunk 4]
```diff
@@ -1,112 +1,100 @@
...
+        dbPositions = dbPositions.Where(p => p.IsActive);
+      }
+
+      return dbPositions.Skip(skipCount).Take(takeCount).ToList();
+    }
+
+    public bool PositionContainsUsers(Guid positionId)
+    {
+      return _provider.PositionUsers
+          .Any(pu => pu.PositionId == positionId && pu.IsActive);
```
Comment (MariyaSin) [inline src/CompanyService.Data/PositionRepository.cs:64]: tab

**Review by Cheephot — APPROVED** [review]

**Review by ndoriomedov — APPROVED** [review]

File: src/CompanyService.Business/Commands/Position/CreatePositionCommand.cs [hunk 5]
```diff
@@ -21,42 +23,65 @@ public class CreatePositionCommand : ICreatePositionCommand
...
-                response.Status = OperationResultStatusType.Conflict;
+                _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.Conflict;
+
+                response.Status = OperationResultStatusType.Failed;
                 response.Errors.Add("The position name already exists");
                 return response;
             }
 
             response.Body = _repository.Create(_mapper.Map(request, _companyRepository.Get().Id));
             response.Status = OperationResultStatusType.FullSuccess;
```
[the remark this answers is not in the document]
Comment (BelehovEgor) [reply src/CompanyService.Business/Commands/Position/CreatePositionCommand.cs:81]: default value null

**Pull request merged**